.github/workflows/full_kubeflow_integration_test.yaml

```
@@ -54,70 +59,84 @@ jobs:
-        kubectl get secret mlpipeline-minio-artifact -n "$KF_PROFILE" -o json | jq -r '.data | keys[] as $k | "\($k): \(. | .[$k] | @base64d)"' | tr '\n' ' '
+
+    - name: Install KNative (if needed)
+      if: always()
+      continue-on-error: true
+      run: |
+        # Check if KNative is required (it's in the kustomization.yaml)
+        if grep -q "knative" example/kustomization.yaml; then
+          echo "KNative detected in kustomization, installing..."
+          if [ -f "tests/gh-actions/install_knative.sh" ]; then
+            ./tests/gh-actions/install_knative.sh
+          elif [ -f "tests/gh-actions/install_knative-cni.sh" ]; then
+            ./tests/gh-actions/install_knative-cni.sh
+          fi
+        else
+          echo "KNative not required, skipping installation"
+        fi
 
     - name: Port Forward Istio Gateway
       run: |
         ingress_gateway_service=$(kubectl get svc --namespace istio-system --selector="app=istio-ingressgateway" --output jsonpath='{.items[0].metadata.name}')
         nohup kubectl port-forward --namespace istio-system svc/${ingress_gateway_service} 8080:80 &
         while ! curl localhost:8080; do echo waiting for port-forwarding; sleep 1; done; echo port-forwarding ready
 
+    # Security scanning with Trivy
+    - name: Run Trivy Security Scan
+      if: always()
+      continue-on-error: true
+      run: |
+        if command -v trivy &> /dev/null; then
+          echo "Running Trivy security scans on critical images..."
+          if [ -f "tests/gh-actions/trivy_scan.py" ]; then
+            python3 tests/gh-actions/trivy_scan.py || true
+          else
+            # Fallback to basic image scan
+            trivy image gcr.io/kubeflow-images-public/centraldashboard || true
+          fi
+        else
+          echo "Trivy not installed properly, skipping security scan"
+        fi
+
+    # Authentication Tests
     - name: Test Auth Integration
       if: always()
       continue-on-error: true
```

I think you can always fail hard
so probably you can remove `
      if: always()
      continue-on-error: true` everywhere. Try to keep it simple for the first iterations.

---

Also removed this